### ISSUE TITLE
The value shown during the keypad input is wrong for frequencies above 999 999 999

### ISSUE THREAD

**Issue by Pmax65** [AA6KL/NanoVNA-H4#3]
Hi Ken.
When the virtual keyboard pop ups the previous value shown at the bottom became wrong for values equal or greater than 1GHz (the formatting cut away the least significant digit).
This should be a very easy issue to fix, and I don't pretend that you fix it soon, you could add this to your next update.

Have a great day.

Massimo

**Comment by AA6KL** [comment]
Thanks Massimo.  This has been fixed at commit 8141f5fdd99d49299cd8891c3412b172e45dc41f.

**Comment by Pmax65** [comment]
Hi Ken,
thank you for your quick fix.
If I'm not wrong you are in Palo Alto, sunny California.
I very love your state and I'm really fascinated of the things that happened at the Silicon Valley in 70s and 80s (even if in Palo Alto those things started years before thanks to Bill Hewlett and Dave Packard). You are lucky living there guy.
I've been there in 2002 and found lovely landscapes and people.

Have a great day.

Thanks again for the fix.

Massimo IK1IZA

**Comment by AA6KL** [comment]
Hi Massimo,

I am travelling between US, Taiwan, and China.  Yes, my US home is at Palo
Alto, CA, a nice place to stay.  However, these days, the living cost here
has become more and more expensive.  It's not the old day's silicone valley
any more.  It have become social media valley, full with Facebook and
Google employees.  We hardware guys are like previous generation geeks....

This NanoVNA is a good one for me to refresh my programming skill, after
working on VLSI design for 20 years.

You have quite a few brilliant suggestions to Hugen.  This is indeed am
excellent community coorperation!

Hope to meet you on the air.

73
Ken, AA6KL

**Comment by Pmax65** [comment]
Hi Ken,
I know and I agree with you about the Silicon Valley story, and I'm aware that we can't do nothing to turn back the time (sigh!). Just to say, few years ago I also bought one Apple II plus to get it here as a souvenir. I had just to clean it a little and turn the 115Vac PSU to our 230Vac domestic line and it worked after more than 30 years... No, honestly after few seconds the EMC capacitor exploded because the ASTEC guys who made that PSU provided a jumper for switching from 115 to 220V but used a 150Vca capacitor in parallel to the inlet :-) 
I think that Steve Wozniak did a great job with that bunch of chips, and it's a disgrace that most of  common people today look at those machines as crap.

I'm older than you, with 33 years of professional HW/FW design in electronics for small companies here in Italy. Mostly regarding industrial control, but I had also an experience as electronic designer for safety grade level 1 redundant radio remote controls, where I had the opportunity to use my experience as an HAM radio. I'm sporadically active in radio. Except for very few QSO at the end of the 80s (more to check my home made three band delta loop antenna than to collect QSL cards) I used (and still use) my free-time to do experiments practically in any field indeed. Just to say, I also designed a full featured optical spectrum analyzer using a PMT and a silicon detector and the optics of an old rotating-mirror-monochromator.
Designing VLSI should be a great professional experience, always at the edge of technology. A good friend of mine works as designer for STMicroelectronics, but he designs chips for digital audio processing. He has nothing to do with the STM32 MCU. 

I must admit that I approached to the nanoVNA with suspicion, I couldn't believe it was performing so good using the odd harmonics method, but thinking a little more the square-wave 1/n harmonics components are not that bad indeed for an handheld VNA. So after I bought some nanoVNA for me and other friends of the radio club I decided to fix some HW incongruences that I found. It was reporting to Hugen about what incongruences, that I also discovered the GitHub community, which (I still agree with you) runs around great people who cooperate each other.

Anyways, I must admit that I feel a little discomfort using the Eclipse IDE (it took me  5 nights to get a working setup, despite at last 90% of my designs involved MCUs programing), it has too too too much configuration properties for me :-) . I still have the parser showing me some inexistent errors indeed and here and there in the code it fails to show the enabled conditional code, but I decided to ignore it.

Thank you for you great job porting the code to the 303 and also for sharing your time with us here.

Hope to meet you on the air too (even if, as said, I'm almost only a listener now)
73
Massimo IK1IZA
